src/Data/PreciseDateTime.purs

```diff
@@ -154,4 +174,13 @@ toDateTimeLossy :: PreciseDateTime -> DateTime
 toDateTimeLossy (PreciseDateTime dt _) = dt
 
 fromDateTime :: DateTime -> PreciseDateTime
-fromDateTime dt = PreciseDateTime dt bottom
```

Comment:
This was just wrong. Nanoseconds would always be 0.

Reply:
DateTime contains millis, how would nanos be anything but 0?

Reply:
I don't think we have a clear invariant for what nanoseconds contains -- in `adjust`, the returned value has all the subseconds in nanoseconds (including millis)

Reply:
I don't think we should be trying to duplicate information in any case. That would be like DateTimes millis contains seconds as well.

Reply:
I guess that explains why I made it `bottom`, but then I did something different in `adjust`.

I think I agree with @natefaubion. It will probably simplify things too.

Reply:
In this case, perhaps `mkPreciseDuration :: DateTime -> Nanoseconds -> PreciseDuration` would make sense?

Reply:
I think `PreciseDuration` is fine as-is, and maybe you meant `mkPreciseDateTime`.

I think we should avoid `mkPreciseDateTime :: DateTime -> Nanosecond -> PreciseDateTime` for the reason you gave in https://github.com/awakesecurity/purescript-precise-datetime/issues/9#issuecomment-369682379 🙂